TCAT/Controllers/RouteDetail+ContentViewController.swift

```diff
@@ -173,23 +162,33 @@ class RouteDetailContentViewController: UIViewController, GMSMapViewDelegate, CL
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
 
-        if let timer = networkTimer {
+        // Live Tracking Network Timer
+        
+        if let timer = liveTrackingNetworkTimer {
             timer.invalidate()
         }
-
-        networkTimer = Timer.scheduledTimer(timeInterval: networkRefreshRate, target: self, selector: #selector(getBusLocations),
+        liveTrackingNetworkTimer = Timer.scheduledTimer(timeInterval: liveTrackingNetworkRefreshRate, target: self, selector: #selector(getBusLocations),
                                             userInfo: nil, repeats: true)
-        networkTimer!.fire()
+        liveTrackingNetworkTimer!.fire()
+        
+        // Bus Delay Network Timer
+        
+        if let timer = busDelayNetworkTimer {
+            timer.invalidate()
+        }
+        busDelayNetworkTimer = Timer.scheduledTimer(timeInterval: busDelayNetworkRefreshRate, target: self, selector: #selector(getDelays),
+                                                    userInfo: nil, repeats: true)
+        busDelayNetworkTimer!.fire()
 
     }
     
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
-        networkTimer?.invalidate()
-        networkTimer = nil
-        UIApplication.shared.statusBarStyle = .default
-        banner?.dismiss()
-        banner = nil
+        liveTrackingNetworkTimer?.invalidate()
+        liveTrackingNetworkTimer = nil
```

Comment:
is it necessary to set it to nil? after you invalidate it its pretty much nil. just got to make sure before you use it again to set liveTrackingNetworkTimer to a new Timer